core/src/main/java/com/linecorp/armeria/common/Scheme.java

```diff
@@ -78,8 +78,12 @@
         if (scheme == null) {
             return Optional.empty();
         }
-
-        return Optional.ofNullable(SCHEMES.get(Ascii.toLowerCase(scheme)));
+        final String lowercaseScheme = Ascii.toLowerCase(scheme);
+        final Optional<Scheme> parsedScheme = Optional.ofNullable(SCHEMES.get(lowercaseScheme));
```

Comment:
We could make sure to only create `Optional` once

```java
Scheme parsedScheme = SCHEMES.get(lowercaseScheme);
if (parsedScheme != null) {
  return Optional.of(parsedScheme);
}

return Optional.ofNullable(SCHEMES.get(SerializationFormat.NONE.uriText() + '+' + lowercaseScheme));
```

Reply:
Done